src/main/java/dev/openfeature/javasdk/OpenFeatureClient.java

```diff
@@ -94,9 +95,14 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
             if (details == null) {
                 details = FlagEvaluationDetails.<T>builder().build();
             }
+            if (e instanceof OpenFeatureError) {
+                details.setErrorCode(((OpenFeatureError)e).getErrorCode());
+            } else {
+                details.setErrorCode(ErrorCode.GENERAL);
+            }
+            details.setErrorMessage(e.getMessage());
             details.setValue(defaultValue);
-            details.setReason(Reason.ERROR);
-            details.setErrorCode(e.getMessage());
+            details.setReason(Reason.ERROR.toString());
```

Comment:
Should we consider whether a provider might want to different reason to `ERROR`?

Reply:
The spec says that reason must indicate an error in error cases: https://github.com/open-feature/spec/blob/f74a8d98763bcac5d9a5c67a40ca9d7c2d445180/specification/sections/01-flag-evaluation.md#requirement-148

I suppose the provider could provide a more specific error reason, but I think additional information can be conveyed by the `errorMessage` and `errorCode` fields.